android/src/main/java/com/amplitude/skylab/Variant.java

```diff
@@ -2,39 +2,48 @@
 
 import android.util.Log;
 
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 import org.json.JSONException;
 import org.json.JSONObject;
 
 public class Variant {
-    public final String value;
-    public final Object payload;
 
-    public Variant(String value) {
+    @Nullable public final String value;
+    @Nullable public final Object payload;
+
+    public Variant(@Nullable String value) {
         this(value, null);
     }
 
-    public Variant(String value, Object payload) {
+    public Variant(@Nullable String value, @Nullable Object payload) {
         this.value = value;
         this.payload = payload;
     }
 
+    @Nullable
+    @Deprecated
```

Comment:
The `value()` and `payload()` are redundant since the fields `value` and `payload` are public & final.